symengine/polys.h

```diff
@@ -0,0 +1,70 @@
+/**
+ *  \file polys.h
+ *  Class for sparse Polynomial: Polynomial
+ *
+ **/
+#ifndef SYMENGINE_POLYS_H
+#define SYMENGINE_POLYS_H
+
+#include <symengine/basic.h>
+#include <symengine/dict.h>
+
+namespace SymEngine {
+//! Polynomial Class
+class Polynomial : public Basic{
+public:
+    //! `vars_` : Variables of the Polynomial
+    //! `polys_set_` : holds the Polynomial
+    vec_symbol vars_;
+    hash_set polys_set_;
```

Comment:
Exactly. Then we can just depend on it. We should probably do something similar with regards to the `Expression`, I don't even know if it is possible, but it might.